src/k8s_client/k8s_client.go

```diff
@@ -182,6 +183,11 @@ func (c *k8sClient) DeletePods(namespace string) error {
 	return c.client.CoreV1().Pods(namespace).DeleteCollection(context.TODO(), metav1.DeleteOptions{}, metav1.ListOptions{})
 }
 
+func (c *k8sClient) PatchNamespace(namespace string, data []byte) error {
+	_, err := c.client.CoreV1().Namespaces().Patch(context.TODO(), namespace, types.StrategicMergePatchType, data, metav1.PatchOptions{})
```

Comment:
TODO for our future selves: We should be passing the context to these functions.